components/sup/src/manager/mod.rs

```diff
@@ -156,6 +206,26 @@ impl Manager {
         Self::new(cfg, member, fs_cfg)
     }
 
+    pub fn service_status(cfg: ManagerConfig, ident: PackageIdent) -> Result<ServiceStatus> {
+        let services = Self::status(cfg)?;
+
+        for status in services {
+            if status.package.satisfies(&ident) {
+                return Ok(status);
+            }
+        }
+
+        Err(sup_error!(Error::ServiceNotLoaded(ident)))
+    }
+
+    pub fn status(cfg: ManagerConfig) -> Result<Vec<ServiceStatus>> {
+        let state_path = Self::state_path_from(&cfg);
+        let fs_cfg = FsCfg::new(state_path);
+
+        let dat = File::open(&fs_cfg.services_data_path)?;
+        Ok(serde_json::from_reader(&dat)?)
```

Comment:
Oh I think i see...

You are saying bail on that `From<serde_json::Error>` implementation and actually match here and explicitly `Err` my `sup_error!(Error::SerdeJsonError(err))`

That does sound better